.github/workflows/setupapp.yml

```diff
@@ -66,6 +66,42 @@ jobs:
         fail_ci_if_error: false
         file: ./coverage.xml
 
+  min-dep-py3:  # min dependencies installed
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [windows-latest, macOS-latest, ubuntu-latest]
+    timeout-minutes: 20
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.x
+      uses: actions/setup-python@v1
+      with:
+        python-version: '3.x'
+    - name: Prepare pip wheel
+      run: |
+        which python
+        python -m pip install --upgrade pip wheel
+    - if: runner.os == 'windows'
+      name: Install torch cpu from pytorch.org (Windows only)
+      run: |
+        python -m pip install torch==1.4 -f https://download.pytorch.org/whl/cpu/torch_stable.html
+    - name: Install the dependencies
+      run: |
+        # min. requirements for windows instances
+        python -m pip install torch==1.4
+        python -c "f=open('requirements-dev.txt', 'r'); txt=f.readlines(); f.close(); print(txt); f=open('requirements-dev.txt', 'w'); f.writelines(txt[1:4]); f.close()"
+        cat "requirements-dev.txt"
+        python -m pip install -r requirements-dev.txt
```

Comment:
Should we uninstall all the packages in the full requirements list first?

Reply:
line 94 actually modifies `requirements-dev.txt`, so this line will install 
```
-r requirements.txt
coverage
parameterized
```